File: src/repositories/user_repository.py
from typing import Optional
from src.models.user import User
from .base_repository import BaseRepository

class UserRepository(BaseRepository[User]):
    def find_by_name(self, name: str) -> Optional[User]:
        for user in self.get_all():
            if user.name == name:
                return user
        return None

File: server/src/repositories/user.repository.js
// server/src/repositories/user.repository.js
const User = require('../models/user.model');

class UserRepository {
  async findById(id) {
    return User.findById(id);
  }

  async findByEmail(email) {
    return User.findOne({ email });
  }

  async create(userData) {
    const user = new User(userData);
    return user.save();
  }

  async updateById(id, updateData) {
    return User.findByIdAndUpdate(id, updateData, { new: true });
  }
}

module.exports = new UserRepository();
